### PR TITLE
feature: Toggle backdrop visibility

### LINKED ISSUE #480: Make background image toggle more visible
## Use Case

Requested by Chantelle! The background image toggle is currently buried in the Viewer settings tab. Can we make it more prominent so users who aren't familiar with the app are more likely to toggle it?

A few areas were suggested for this, either in the top section above the canvas (currently contains just the toggle for keeping range) or in the toolbar.

image

This is also related to #303, "redesign viewer settings!"

## Acceptance Criteria

_Please describe how you know this is done_

## Details

- Do users just want to toggle it on and off from here? Will they need to switch between multiple background images?
- Is selecting the first background image in the dataset okay as a default?
- If TFE was a nucmorph-only app, this toggle might read, "Show original images" and also decrease the object opacity when toggled. Are there ideas from this that we can reuse? 
  - What generalizes and what doesn't? (changing object transparency when background images are enabled?)
- Semi-related also to #472, where we could add a visibility toggle for background images

### PR DESCRIPTION
**Pull request by ShrimpCryptid**
Problem
=======
Closes #480, "make background image toggle more visible"!

This change adds a new onscreen toggle for the backdrop image to help make it more discoverable for users. 

*Estimated review size: ~20 minutes, medium*

Solution
========
- Added a backdrop visible flag to the `ViewerConfig` type.
- Synced the new backdrop visibility flag to the URL.
- Changed object opacity so it is only applied when backdrops are visible.
- Updated `SettingsTab` with the new vector config.
  - This change moves object opacity to live under the backdrop settings area.
- Adds a new onscreen toggle for the background image in `CanvasWrapper`.

## Type of change
* New feature (non-breaking change which adds functionality)

Steps to Verify:
----------------
1. Open the preview link: https://allen-cell-animated.github.io/timelapse-colorizer/pr-preview-internal/pr-483/viewer?collection=https%3A%2F%2Fdev-aics-dtp-001.int.allencell.org%2Fassay-dev%2Fusers%2FChantelle%2Fcolorizer%2Fnew_backdrop%2Fexploratory_dataset%2Fcollection.json
2. Hover over the backdrop button on the UI. Try clicking the link to go to the viewer settings. Toggle the button.
4. Open a preview link where backdrops are not included: https://allen-cell-animated.github.io/timelapse-colorizer/pr-preview-internal/pr-483/viewer?collection=https%3A%2F%2Fallencell.s3.amazonaws.com%2Faics%2Fnuc-morph-dataset%2Ftimelapse_feature_explorer_datasets%2Fexploratory_dataset%2Fcollection.json
5. Observe the disabled state.

Screenshots (optional):
-----------------------
**Video Talk-through (🔊):**

https://github.com/user-attachments/assets/af1d9cb6-9438-43be-a7fa-b0265cc4feea

image

image

image

### CODE REVIEW

File: src/components/CanvasWrapper.tsx [hunk 1]
```diff
@@ -30,19 +31,32 @@ const RIGHT_CLICK_BUTTON = 2;
...
+                  {text}
+                </p>
+              ))}
+          </>
+        }
+        getPopupContainer={() => divRef.current ?? document.body}
+      >
+        {props.children}
+      </Tooltip>
+    </div>
```
Comment (ShrimpCryptid) [inline src/components/CanvasWrapper.tsx:59]: Recommend reviewing this with whitespace off!

**Review by interim17 — APPROVED** [review]

File: src/components/CanvasWrapper.tsx [hunk 2]
```diff
@@ -225,10 +240,22 @@ export default function CanvasWrapper(inputProps: CanvasWrapperProps): ReactElem
 
   // Update backdrops
   useMemo(() => {
-    canv.setBackdropKey(props.selectedBackdropKey);
-    canv.setBackdropBrightness(props.config.backdropBrightness);
-    canv.setBackdropSaturation(props.config.backdropSaturation);
-  }, [props.selectedBackdropKey, props.config.backdropBrightness, props.config.backdropSaturation]);
+    if (props.selectedBackdropKey !== null && props.config.backdropVisible) {
```
Comment (rugeli) [inline src/components/CanvasWrapper.tsx:243]: This is a nice and important check to have!

**Review by rugeli — APPROVED** [review]
LGTM! Thank you for the demo and the always-helpful comments.

I noticed that clicking outside the checkbox (while hovering over the same row) still toggles it in `Viewer Settings`. This feels like something a user might accidentally trigger. Are we planning to address this in upcoming PRs?

**Comment by ShrimpCryptid** [comment]
> LGTM! Thank you for the demo and the always-helpful comments.
> 
> I noticed that clicking outside the checkbox (while hovering over the same row) still toggles it in `Viewer Settings`. This feels like something a user might accidentally trigger. Are we planning to address this in upcoming PRs?

It's semi-addressed in issue #472, but it's on hold for now pending a redesign of the settings area in #486.

**Pull request merged**